File: lib/crypto/src/main/java/com/example/toysapp/crypto/Base64Util.java
package com.example.toysapp.crypto;

import com.example.toysapp.logger.Logger;

import org.apache.commons.codec.binary.Base64;

public class Base64Util {

    private final Logger logger;

    public Base64Util(Logger logger) {
        this.logger = logger;
    }

    public String encodeToBase64(String text) {
        logger.log("EncoderUtil - encodeToBase64");

        Base64 base64 = new Base64();
        byte[] encodedBytes = base64.encode(text.getBytes());
        return new String(encodedBytes);
    }
}

File: lib/crypto/src/main/java/com/example/toysapp/crypto/Base64Util.kt
package com.example.toysapp.crypto

import com.example.toysapp.logger.Logger
import org.apache.commons.codec.binary.Base64

class Base64Util(
    private val logger: Logger
) {

    fun encodeToBase64(text: String): String {
        logger.log("EncoderUtil - encodeToBase64")

        val base64 = Base64()
        val encodedBytes = base64.encode(text.toByteArray())
        return String(encodedBytes)
    }
}
